src/index.ts

```diff
@@ -243,8 +252,9 @@ const transformWithOptions = async (
       swcrc: false,
       configFile: false,
       sourceMaps: true,
+      ...(options.env ? { env: options.env } : {}),
       jsc: {
-        target,
+        ...(options.env ? {} : { target }),
```

Comment:
Setting `env` is basically the fine-tuned replacement for `target`.

So the behavior here is to use env when set by the users, otherwise use the build target as usual.

Using both is contradictory and [not supported](https://swc.rs/docs/configuration/compilation#env)